distributed/scheduler.py

```diff
@@ -503,9 +503,18 @@ def cleanup(self):
         self.status = 'closing'
         logger.debug("Cleaning up coroutines")
 
-        for w, comm in list(self.worker_comms.items()):
-            with ignoring(AttributeError):
-                yield comm.close()
+        close_nannies = {w: self.rpc(self.get_worker_service_addr(w, 'nanny')).terminate()
```

Comment:
Yes, I can see how that might be cleaner.  Looking into this further it looks like the worker *already* tells the nanny to close by default.